api4/post.go

```diff
@@ -300,6 +301,13 @@ func searchPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	posts, err := c.App.SearchPostsInTeam(terms, c.Session.UserId, c.Params.TeamId, isOrSearch)
 
+	for postID, post := range posts.Posts {
+		if !c.App.SessionHasPermissionToChannel(c.Session, post.ChannelId, model.PERMISSION_READ_CHANNEL) {
```

Comment:
The problem with this is that you are removing results from the resolved query, this means that if you want 100 results, you can get 70 or 40, or 0, when you actually have valid results. We can try to do a while over it, until we fill the block of results, but if we include pagination after that, we have to worry about the offset to, could be a pain in performance.

Reply:
I was thinking in something more "SQL based", but I'm 0/5 about if that is a good solution, because we can generate a very complex query to express the permissions match.

Reply:
@jespino I think you are correct because the [elasticsearch query limits the results to 100](https://github.com/mattermost/enterprise/blob/advanced-permissions-phase-1/elasticsearch/elasticsearch.go#L255). Looks like we will be force to go more complicated.

Reply:
Yes, we can make a simple approach with a while loop, and check how it performs, and how we can improve. Other option is solve it inside mysql and elasticsearch.